test/unit/quality-check/get-changed-quality-checks.test.js

```diff
@@ -0,0 +1,76 @@
+const db = require('../../../app/data')
+const { SCHEME_ID_SFI } = require('../../data/scheme-id')
+const { getChangedQualityChecks } = require('../../../app/quality-check')
+describe('Get changed quality check tests', () => {
+  let paymentRequest
+
+  const resetTables = async () => {
+    await db.manualLedgerPaymentRequest.truncate({ cascade: true, restartIdentity: true })
+    await db.paymentRequest.truncate({ cascade: true, restartIdentity: true })
+    await db.scheme.truncate({ cascade: true })
+  }
+
+  beforeEach(async () => {
+    await resetTables()
+    const scheme = { schemeId: 1, name: 'SFI' }
+
+    paymentRequest = {
+      paymentRequestId: 1,
+      schemeId: SCHEME_ID_SFI,
+      frn: 1234567890,
+      categoryId: 2
+    }
+
+    await db.scheme.create(scheme)
+    await db.paymentRequest.create(paymentRequest)
+    await db.paymentRequest.create({ ...paymentRequest, paymentRequestId: 2 })
+    await db.paymentRequest.create({ ...paymentRequest, paymentRequestId: 3 })
+    await db.paymentRequest.create({ ...paymentRequest, paymentRequestId: 4 })
+    await db.paymentRequest.create({ ...paymentRequest, paymentRequestId: 5 })
+    await db.manualLedgerPaymentRequest.create({ paymentRequestId: 1, ledgerPaymentRequestId: 2, active: false, original: true })
+    await db.manualLedgerPaymentRequest.create({ paymentRequestId: 1, ledgerPaymentRequestId: 3, active: true, original: false })
+    await db.manualLedgerPaymentRequest.create({ paymentRequestId: 4, ledgerPaymentRequestId: 5, active: true, original: false })
+  })
+
+  afterAll(async () => {
+    await resetTables()
+    await db.sequelize.close()
+  })
+
+  test('should return hasDismissed as yes with paymentRequestId of 1', async () => {
+    const qualityChecks = [
+      {
+        frn: 1234567890,
+        paymentRequest: {
+          paymentRequestId: 1
+        }
+      }]
+    const changedQualityCheck = await getChangedQualityChecks(qualityChecks)
+    console.log(changedQualityCheck)
+    expect(changedQualityCheck[0].hasDismissed).toBe('Yes')
+  })
+  test('should return hasDismissed as no with paymentRequestId of 5', async () => {
+    const qualityChecks = [
+      {
+        frn: 1234567890,
+        paymentRequest: {
+          paymentRequestId: 5
+        }
+      }]
+    const changedQualityCheck = await getChangedQualityChecks(qualityChecks)
+    console.log(changedQualityCheck)
+    expect(changedQualityCheck[0].hasDismissed).toBe('No')
+  })
+  test('should return hasDismissed as no with paymentRequestId of 4 and "active" is true', async () => {
```

Comment:
This test isn't testing that "active" is true

Reply:
I think this okay as line 32 sets `active` to `true` for a manual ledger record with a `paymentRequestID` of 4

Reply:
Aaaaahhh I gotcha now -- yup, looking up at the `beforeEach` helped.

Hhmmmmmm I would reword these test cases so tht they reference the state that the payment record is in

Instead of `should return hasDismissed as yes with paymentRequestId of 1'` because this test would fail if you `beforeEach` didn't inc. `paymentRequestId` 1 ya'know? It isn't dependent on that.

What it is dependent on is the fact that `paymentRequestId` 1 exists within `paymentRequest` table and that matching record exists within `manualLedgerPaymentRequest` with `active` equal `false`

So I'd consider something like: `should return hasDismissed as "Yes" (being picky over casing and exact return value heheh) when paymentRequest exists and matching manualLedgerPaymentRequest exists with active set to false`

This name doesn't care what the `paymentRequestId` is (becuz we make tht up in our mock data) but it gives context to the state of this particular record and particular test scenario -- which gives way to asking more questions like, "What about when there isn't a matching `paymentRequest` in `manualLedgerPaymentRequest`, e.g. `paymentRequestId` 2 in your `beforeEach`?"

Reply:
updated test names as suggested.
Considered if additional test needed to cover above point of _What about when there isn't a matching paymentRequest in manualLedgerPaymentRequest, e.g. paymentRequestId 2_ 
However i think this is covered by second test on line 52 where there is a record in `paymentRequest` table but no record in `manualLedgerRequest` table but for `paymentRequestId` 5. Therefore no additional tests added